agent/rpi_helper.py

```diff
@@ -144,3 +145,43 @@ def auto_upgrades_enabled():
             '${distro_id}:${distro_codename}-security' in allowed_origins
     else:
         return False
+
+
+def kernel_cmdline():
+    """
+    Parses kernel parameters (aka cmdline).
+    :return: A dict where 'name' is kernel parameter name and 'value' is its value or empty string if no value provided.
+    """
+    cmdline_path = Path('/proc/cmdline')
+    cmdline_matches = re.compile(r"([\w\-\.]+)(\=(\"[\w\W]+\"|[\w\S]+)?)?").findall(cmdline_path.read_text())
```

Comment:
I recommend to move the re pattern compilation out of the function and use already compiled instance in it.

Reply:
It is going to only be used twice during the whole lifetime of the program. And it's going to parse a string which is around 100 chars long. No need to optimize anything.

Reply:
As you wish. But fixing this would take you less time than writhing the reply ^.
The meaning of re pattern compilation is to do it once and use multiple times. If you believe it's useless - them you can do `findall` without compilation at all